buildSrc/src/main/kotlin/com/datadog/gradle/plugin/benchmark/ReviewBenchmarkResultsTask.kt

```diff
@@ -45,16 +46,31 @@ open class ReviewBenchmarkResultsTask : DefaultTask() {
 
     // region Internal
 
-    private fun processDeviceBenchmarks(deviceDir: File) {
+    private fun processBenchmarksPerDevice(
+        deviceDir: File
+    ) {
         println("processing benchmark results for device ${deviceDir.name}")
         if (deviceDir.exists() && deviceDir.isDirectory && deviceDir.canRead()) {
+            val benchmarksResults = mutableMapOf<String, Long>()
             deviceDir.listFiles()?.forEach {
                 if (it.isFile && it.canRead() && it.extension == "json") {
-                    check(visitor.visitBenchmarkJsonFile(it, extension.thresholds, extension.ignored)) {
-                        "One or more benchmark tests in ${it.path} didn't match the expected threshold."
-                    }
+                    visitor.visitBenchmarkJsonFile(it, benchmarksResults)
```

Comment:
You should verify each strategy within this loop, otherwiser, you will only test against the last parsed file.

Reply:
Actually not...what I am doing there is accumulating all the benchmarks for one device at a time in the **bechmarkResults** and then validating those through the strategies.

Reply:
Oh yes you're right